rotkehlchen/tests/api/test_assets.py

```diff
@@ -262,3 +262,150 @@ def test_ignored_assets_endpoint_errors(rotkehlchen_api_server_with_exchanges, m
+
+
+def test_get_assets_mappings(rotkehlchen_api_server):
+    """Test that providing a list of asset identifiers, the appropriate assets mappings are returned."""  # noqa: E501
+    queried_assets = ('BTC', 'TRY', 'EUR')
+    response = requests.post(
+        api_url_for(
+            rotkehlchen_api_server,
+            'assetsmappingresource',
+        ),
+        json={'identifiers': queried_assets},
+    )
+    result = assert_proper_response_with_result(response)
+    assert len(result) == 3
+    for entry in result:
+        assert entry['identifier'] in queried_assets
+
+    # check that providing multiple order_by_attributes fails
+    response = requests.post(
+        api_url_for(
+            rotkehlchen_api_server,
+            'assetsmappingresource',
+        ),
+        json={'identifiers': ['BTC', 'TRY', 'invalid']},
+    )
+    assert_error_response(response, contained_in_msg='One or more identifiers do not exist')
+
+
+def test_search_assets(rotkehlchen_api_server):
+    """Test that searching for assets using a keyword works."""
+    response = requests.post(
+        api_url_for(
+            rotkehlchen_api_server,
+            'assetssearchresource',
+        ),
+        json={
+            'substring_search': 'Bitcoin',
+            'limit': 50,
+            'order_by_attributes': ['name'],
+            'ascending': [True],
+        },
+    )
+    result = assert_proper_response_with_result(response)
+    for entry in result:
+        assert 'bitcoin' in entry['name'].lower() or entry['symbol'].lower()
+
+    # use a different keyword
+    response = requests.post(
+        api_url_for(
+            rotkehlchen_api_server,
+            'assetssearchresource',
+        ),
+        json={
+            'substring_search': 'eth',
+            'limit': 50,
+            'order_by_attributes': ['name'],
+            'ascending': [True],
+        },
+    )
+    result = assert_proper_response_with_result(response)
+    for entry in result:
+        assert 'eth' in entry['name'].lower() or entry['symbol'].lower()
+
+    # check that searching for a non-existent asset returns nothing
+    response = requests.post(
+        api_url_for(
+            rotkehlchen_api_server,
+            'assetssearchresource',
+        ),
+        json={
+            'substring_search': 'idontexist',
+            'limit': 50,
+            'order_by_attributes': ['name'],
+            'ascending': [True],
+        },
+    )
+    result = assert_proper_response_with_result(response)
+    assert len(result) == 0
+
+
+def test_search_assets_with_levenshtein(rotkehlchen_api_server):
+    """Test that searching for assets using a keyword works(levenshtein approach)."""
+    response = requests.post(
+        api_url_for(
+            rotkehlchen_api_server,
+            'assetssearchlevenshteinresource',
+        ),
+        json={
+            'substring_search': 'Bitcoin',
```

Comment:
isn't there a good example search to try that shows the power of fuzzy search? Another good of this approach is you don't need to speciy name/symbol, this search just finds multiple matches in either.